### PR TITLE
support for starttls inner socket in UpstreamProxyProtocolSocket

### PR DESCRIPTION
**Pull request by VishalDamgude**
Commit Message: support for starttls inner socket in UpstreamProxyProtocolSocket
Additional Description: When starttls socket is configured as inner socket in UpstreamProxyProtocolSocket for the cluster, we are unable to switch to TLS. When an L7 filter (e.g SMTP) invokes startSecureTransport() on UpstreamProxyProtocolSocket as a result of STARTTLS command, the api returns false due to missing implementation.

Risk Level: Low
Testing: Unit Tests, Local tests with SMTP filter.
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] https://github.com/envoyproxy/envoy/issues/26592
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by alyssawilk** [comment]
Huh, my initial pass was going to be to ask we add this directly to the passthrough base class but there's a comment that should not be done.
I'm going to throw this over to @cpakulski and @lizan because I don't understand why that was the case in https://github.com/envoyproxy/envoy/pull/13112

**Comment by yanavlasov** [comment]
It would be good to add an integration test that validates this plumbing end to end.

**Comment by cpakulski** [comment]
After reviewing the code, I think the call to `startSecureTransport` should be added to base class `PassthroughSocket`. Yes, there is a comment I added saying that _"startSecureTransport method should not be called for this transport socket."_, but I really meant _"so far nobody uses it so it is not implemented"_. All transport sockets derived from PassthroughSocket does not work if "inner" socket is StartTLS: proxy and tap.

Also, integration tests should be added as mentioned above.

**Review by cpakulski — CHANGES_REQUESTED** [review]
Please move `startSecureTransport` to base class.

File: test/extensions/transport_sockets/proxy_protocol/proxy_protocol_test.cc [hunk 1]
```diff
@@ -723,6 +724,52 @@ TEST_F(ProxyProtocolSocketFactoryTest, ImplementsSecureTransportCallInnerFactory
   ASSERT_TRUE(factory_->implementsSecureTransport());
 }
 
+// Test startSecureTransport() api call on UpstreamProxyProtocolSocket
+TEST_F(ProxyProtocolTest, TestStartSecureTransportCall) {
+  ProxyProtocolConfig config;
+  config.set_version(ProxyProtocolConfig_Version::ProxyProtocolConfig_Version_V2);
+  initialize(config, nullptr);
+
+  EXPECT_CALL(*inner_socket_, startSecureTransport()).WillOnce(Return(false));
```
[the remark this answers is not in the document]
Comment (VishalDamgude) [reply test/extensions/transport_sockets/proxy_protocol/proxy_protocol_test.cc:733]: sure, will address this.

**Comment by ravenblackx** [comment]
@alyssawilk Review SLO ping.

**Pull request closed without merging**